### PR TITLE
Extend weight compression with INT8 symmetric scheme

### PR DESCRIPTION
**Pull request by l-bat**
### Changes

Added `INT8_SYM` compression mode

### Reason for changes

 `INT8_SYM` mode can provide a better performance and required for dynamic quantization

### Related tickets

124823

### Tests

Updated tests/openvino/native/quantization/test_weights_compression.py

### CODE REVIEW

File: docs/compression_algorithms/CompressWeights.md [hunk 1]
```diff
@@ -79,7 +87,7 @@ Here is the perplexity and model size before and after weight compression for di
   </tr>
   <tr>
     <td class="tg-0pky">databricks/dolly-v2-3b</td>
-    <td class="tg-0pky">int8</td>
+    <td class="tg-0pky">int8_asym</td>
```
Comment (l-bat) [inline docs/compression_algorithms/CompressWeights.md:90]: @AlexKoff88 Should we provide results for `int8_sym` mode on these models?

File: nncf/torch/quantization/quantize_model.py [hunk 2]
```diff
@@ -104,8 +106,8 @@ def compress_weights_impl(
     """
     if ignored_scope is not None:
         raise AttributeError("Torch backend does not support ignored scope.")
-    if mode != CompressWeightsMode.INT8:
-        raise AttributeError(f"Torch backend supports only INT8 mode for weight compression, but given {mode} mode.")
+    if mode != CompressWeightsMode.INT8_ASYM:
```
Comment (l-bat) [inline nncf/torch/quantization/quantize_model.py:109]: @AlexKoff88 Is it required to add `INT8_SYM` mode for Torch backend?

File: docs/compression_algorithms/CompressWeights.md (hunk 1 above)
Comment (l-bat) [reply docs/compression_algorithms/CompressWeights.md:90]: No (discussed offline)

File: nncf/torch/quantization/quantize_model.py (hunk 2 above)
Comment (l-bat) [reply nncf/torch/quantization/quantize_model.py:109]: No (discussed offline)

File: nncf/quantization/algorithms/weight_compression/openvino_backend.py [hunk 3]
```diff
@@ -103,12 +103,12 @@ def do_compression(
...
         internal_weight_params = all_weight_params
-        if mode != CompressWeightsMode.INT8:
+        if mode not in [CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM]:
             internal_weight_params = list(filter(lambda wp: wp.metatype != OVEmbeddingMetatype, all_weight_params))
             if not is_last_layer_compressed:
                 internal_weight_params = internal_weight_params[:-1]
-            primary_config = WeightCompressionConfig(mode=mode, group_size=group_size)
-            _assign_mixed_precision(internal_weight_params, ratio, primary_config)
+        primary_config = WeightCompressionConfig(mode=mode, group_size=group_size)
+        _assign_mixed_precision(internal_weight_params, ratio, primary_config)
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/openvino_backend.py:111]: Could please refactor this part to make it more clear?
- `_assign_mixed_precision` is not actually about mixed precision when ratio=1, I'd move outside this logic
- internal weight params and if condition are needed for mixed precision only

BTW, which mode is assigned for embeddings in INT4_ASYM / INT4_SYM? Is it always INT8_ASYM? I'd mention this in docs.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: nncf/quantization/algorithms/weight_compression/openvino_backend.py [hunk 4]
```diff
@@ -212,7 +204,10 @@ def _do_integer_quantization(
     """
     The method quantizes the given weights to integer data type in accordance with the compression config.
     The config defines a quantization mode:
-        INT8 mode refers to unsigned int8 asymmetric weight compression - quantization to [0, 255] range.
+        INT8_SYM mode refers to unsigned int4 symmetric weight compression with a fixed zero point equals to 128 -
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/openvino_backend.py:207]:
```suggestion
        INT8_SYM mode refers to unsigned int8 symmetric weight compression with a fixed zero point equals to 128 -
```

**Review by ljaljushkin — APPROVED** [review]
Special thanks for refactoring and fixing issue with printing statistics

**Review by andreyanufr — APPROVED** [review]

File: nncf/parameters.py [hunk 5]
```diff
@@ -62,20 +62,28 @@ class DropType(Enum):
...
+    :param INT8: Mode is deprecated and will be removed in future releases. Please use `INT8_ASYM` instead.
     """
 
-    INT8 = "int8"
+    INT8_SYM = "int8_sym"
+    INT8_ASYM = "int8_asym"
     INT4_SYM = "int4_sym"
     INT4_ASYM = "int4_asym"
     NF4 = "nf4"
+    INT8 = "int8"  # Deprecated mode
```
Comment (alexsu52) [inline nncf/parameters.py:89]: What do you think about leaving INT8 as an alias for INT8_ASYM?

**Pull request merged**